cmd/sync-tags/gomod.go

```diff
@@ -59,10 +59,14 @@ func updateGomodWithTaggedDependencies(tag string, depsRepo []string) (bool, err
 			return changed, fmt.Errorf("failed to get tag %s for %q: %v", tag, depPkg, err)
 		}
 		rev := commit.String()
-		pseudoVersion := fmt.Sprintf("v0.0.0-%s-%s", commitTime.UTC().Format("20060102150405"), rev[:12])
+		pseudoVersionOrTag := fmt.Sprintf("v0.0.0-%s-%s", commitTime.UTC().Format("20060102150405"), rev[:12])
```

Comment:
as soon as we start adding tags, v0.0.0 will no longer be the correct pseudoversion for commits in between semver tags... see https://tip.golang.org/cmd/go/#hdr-Pseudo_versions

My reading of that means that when generating a non-tagged version:
* if the nearest semver tag is `v0.17.0`, we should generate a pseudoversion using the format string `v0.17.1-0.%s-%s`
* if the nearest semver tag is `v1.17.0-alpha.3`, we should generate a pseudoversion using the format string `v1.17.0-alpha.3.0.%s-%s`

Reply:
isn‘t there automatic normalization through go mod we can use?

Reply:
that means we need to know the nearest existing tag for the current repo... not sure whether we've fetched that info and if it is available currently

Reply:
> isn‘t there automatic normalization through go mod we can use?

I'm not sure how much of that is available to us before the commits/tags are pushed to the authoritative VCS

Reply:
It isn't available currently, but once we fetch that looks like we can plug it into https://github.com/golang/vgo/blob/master/vendor/cmd/go/internal/modfetch/pseudo.go#L48

Reply:
Updated to not use pseudoversions anymore, so this problem is fixed.

Reply:
but tags? why didn't we come up with that in the first place? :)